src/replication.c

```diff
@@ -1105,6 +1385,9 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd)
 
 	replica->pool_guid = ack_data->pool_guid;
 	replica->zvol_guid = ack_data->zvol_guid;
+	replica->replica_id = malloc(strlen(ack_data->replica_id)+1);
```

Comment:
updated to `replica->replica_id = malloc(sizeof(char) * (strlen(ack_data->replica_id) + 1))`